src/algorithms/stats/histogram.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2006-2016  Music Technology Group - Universitat Pompeu Fabra
+ *
+ * This file is part of Essentia
+ *
+ * Essentia is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation (FSF), either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the Affero GNU General Public License
+ * version 3 along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+
+#ifndef ESSENTIA_HISTOGRAM_H
+#define ESSENTIA_HISTOGRAM_H
+
+#include "algorithm.h" 
+
+namespace essentia {
+namespace standard {
+
+class Histogram : public Algorithm {
+
+ protected:
+  Input<std::vector<Real> > _array;
+  Output<std::vector<Real> > _binCounts;
+  Output<std::vector<Real> > _binRanges;
+
+  std::string  _normalize;
+  Real _minValue;
+  Real _maxValue;
+  int  _numberBins;
+
+ public:
+  Histogram() {
+    declareInput(_array, "array", "the input array");
+    declareOutput(_binCounts, "binCounts", "the values in the equally-spaced bins");
```

Comment:
The name ```binCounts``` is not the best one because it does not make sense for the normalized histogram. Perhaps, rename to "histogram"?